internal/supervisor/import.go

```diff
@@ -277,48 +274,43 @@ func (s *ImportSupervisor) transitionImportInProgress(imp *model.Import, install
 	return model.ImportStateComplete
 }
 
-// transitionImportComplete handles the transition for an import in the 'complete' state.
-// It performs final adjustments and cleanup after the import is done.
+// transitionImportComplete handles the transition for an import in the 'complete'
+// state. It performs final adjustments and cleanup after the import is done.
 func (s *ImportSupervisor) transitionImportComplete(imp *model.Import, installation *cloud.InstallationDTO, logger log.FieldLogger) string {
 	if installation.State != cloud.InstallationStateStable {
 		logger.Debug("Waiting for installation to be stable")
 		return imp.State
 	}
 
-	logger.Debug("Installation is Stable")
-
-	var adjustmentRequired bool
-	size := installation.Size
-	if size == model.Size1000String {
-		logger.Infof("Resizing installation to %s", model.SizeCloud10Users)
-		size = model.SizeCloud10Users
-		adjustmentRequired = true
-	}
-
-	s3TimeoutEnv := installation.PriorityEnv[model.S3EnvKey]
-	if s3TimeoutEnv.Value == fmt.Sprintf("%d", model.S3ExtendedTimeout) {
-		logger.Info("Setting S3 timeout back to the default value")
-		s3TimeoutEnv.Value = fmt.Sprintf("%d", model.S3DefaultTimeout)
-		adjustmentRequired = true
-	}
-
-	if !adjustmentRequired {
-		logger.Info("No adjustments required")
+	logger.Info("Running post-import installation configuration check")
+	patch := getPostImportPatch(installation.Installation, logger)
+	if patch == nil {
+		logger.Info("No installation adjustments required")
 		if imp.Error != "" {
 			return model.ImportStateFailed
 		}
 		return model.ImportStateSucceeded
 	}
 
-	logger.Info("Adjustments required")
-	patchInstallation := &cloud.PatchInstallationRequest{
-		Size: &size,
-		PriorityEnv: cloud.EnvVarMap{
-			model.S3EnvKey: s3TimeoutEnv,
-		},
+	logger.Info("Adjusting installation configuration")
+
+	var err error
+	if installation.APISecurityLock {
+		err = s.cloud.UnlockAPIForInstallation(installation.ID)
+		if err != nil {
+			logger.WithError(err).Error("Failed to unlock installation")
+			return imp.State
+		}
+
+		defer func() {
+			err = s.cloud.LockAPIForInstallation(installation.ID)
+			if err != nil {
+				logger.WithError(err).Error("Failed to relock installation")
+			}
+		}()
 	}
```

Comment:
Nit: Seen this twice, could write a function to wrap the update with a lock/unlock?

Reply:
Yep, good idea. Update pushed.